src/models/users.js

```diff
@@ -9,22 +6,65 @@ import {
 
 // Supported authentication methods.
 const authMethods = {
-  BASIC: 'basic'
+  BASIC: ({ username, password }) => {
+    // For now we only support admin authentication.
+    if (username !== 'admin' || password !== config.get('adminPass')) {
+      return Promise.reject(new Error(UNAUTHORIZED));
+    }
+
+    return Promise.resolve({
+      id: 'admin',
+      scope: 'admin'
+    });
+  },
+
+  AUTH_PROVIDER: (data) => {
+    return Promise.resolve({
+      id: data,
+      scope: 'user'
+    });
+  },
+};
+
+module.exports = (sequelize, DataTypes) => {
+  const User = sequelize.define('Users', {
+    opaqueId: { type: DataTypes.STRING(256) },
+    provider: { type: DataTypes.STRING(32) },
+  }, {
+    indexes: [{
+      unique: true,
+      fields: ['opaqueId', 'provider', 'ClientKey']
```

Comment:
It's created automatically by sequelize with the association we create in `associate`.

I agree this is not really reader-friendly because the association is done quite a few lines below. A comment will definitely help here.

I can also configure explicitely the foreign key so that we control the name, and then we can have a lowercase as first letter. Do you think it's better even with the comment ?